lib/api.js

```diff
@@ -967,11 +967,8 @@ function startOrContinueTrace (xtrace, build, run, opts, cb) {
 
   // If already tracing, continue the existing trace ignoring any xtrace
   // passed as the first argument, unless forcing a new trace.
-  let last;
-  if (!opts.forceNewTrace) {
-    last = Span.last
-  }
-  if (last) {
+  const last = Span.last;
+  if (last && !opts.forceNewTrace) {
```

Comment:
If forcing a new trace, there is no need to "clean up" any existing `Span.last` here because it will be reset when calling `Span.makeEntrySpan()` later in the code?

Also later in line 999 there is the comment in the call to finalize:
> no last or runInstrument() would already have been called.

... would that be affected by the force new trace logic?

Reply:
there is no need to clean up any existing `Span.last` because a completely new context is created that will not inherit from the existing context (so it won't know anything about this particular `Span.last`). likewise, even if `last` was around but we're forcing a trace then that last will be associated with a non active context.

each context can be thought of as nested babushkas. every time a new span is created a new outer babushka is created. but until a `Span.last` or `Event.last` is created on the outer babushka the context keeps searching the inner nested babushkas. those are never overwritten by an lower level span; they're just masked behind an outer babushka's copy of the same name, e.g. `Span.last`.

when a completely new context is created it's like creating a whole new set of babushka's but with only the inner-most one to start with.